lib/_components/appCard/template.njk

```diff
@@ -1,31 +1,21 @@
-{% from "appBadges/macro.njk" import appBadges %}
+{% from 'card/macro.njk' import card %}
+{% from "./macro.njk" import cardInfo, cardHeader %}
 {% if params.items.length > 0 %}
-	<ul class="nhsuk-grid-row nhsuk-card-group{%- if params.classes %} {{ params.classes }}{% endif %}">
-		{% for item in params.items | sortByOrder(params.order) %}
+	<ul class="nhsuk-grid-row nhsuk-card-group">
+	{% for item in params.items | sortByOrder(params.order) %}
 		{% if (not item.data.status or params.page_status == 'DRAFT' or item.data.status != 'DRAFT') and params.page.url != item.url %}
-		<li class="nhsuk-grid-column-one-third nhsuk-card-group__item">
-			<div class="nhsuk-card nhsuk-card--clickable">
-				<div class="nhsuk-card__content">
-					<div class="nhsuk-card__header">
-						<h2 class="nhsuk-card__heading nhsuk-heading-m">
-							<a class="nhsuk-card__link" href="{{ item.url | url | pretty }}">{{ item.data.title }}</a>
-						</h2>
-						{{ appBadges({
-								badges: item.data.badges
-							}) 
-						}}
-					</div>
-					{% if item.data.description %}
-					<p class="nhsuk-card__description">{{ item.data.description | markdown("inline") | safe }}</p>
-					{% endif %}
-					{% if item.date %}
-					<time class="nhsuk-body-s nhsuk-u-secondary-text-color nhsuk-u-margin-top-5 nhsuk-u-margin-bottom-0" datetime="{{ item.date | date }}">{{ item.date | date("d LLLL y") }}</time>	
-					{% endif %}
-				</div>
-			</div>
+		<li class="{% if content | toc %} nhsuk-grid-column-one-half {% else %} nhsuk-grid-column-one-third {% endif %} nhsuk-card-group__item">
```

Comment:
Not sure about this condition (and possibly same condition in other macros/templates)

We have the decision to display `hero` decided in the page itself. Perhaps we need a similar approach here. So either decide to show toc based on page frontmatter, or by the layout that calls the macros. This decouples the decision from a macro - it's just rendering what its been asked, rather than makiong decisions based on content.

What do you think?

Reply:
@yantantether 
I like the decision to split the logic from the macro. 
However, the card styling itself needs the decision to determine its width. So if we split out the decision into the page, like the hero. We'd still pass something into the macro to determine which width to use - which could be used in multiple places. 

In addition, this check is only used here for this specific component which I think should be contained within the macro. It might be worth looking into a TOC redesign of how it works / is shown or interacted with on the page?